atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientImpl.java

```diff
@@ -124,10 +233,50 @@ public CqlResult execute_cql3_query(CqlQuery cqlQuery,
             TException {
         ByteBuffer queryBytes = ByteBuffer.wrap(cqlQuery.toString().getBytes(StandardCharsets.UTF_8));
 
-        return client.execute_cql3_query(queryBytes, compression, consistency);
+        return executeMethod(() -> client.execute_cql3_query(queryBytes, compression, consistency));
     }
 
     private ColumnParent getColumnParent(TableReference tableRef) {
         return new ColumnParent(AbstractKeyValueService.internalTableName(tableRef));
     }
+
+    private <T> T executeMethod(ThrowingSupplier<T> supplier)
+            throws InvalidRequestException, UnavailableException, TimedOutException, SchemaDisagreementException, TException {
+        try {
+            return supplier.apply();
+        } catch (Exception e) {
+            isValid = blackListedExceptions.stream()
```

Comment:
I don't agree with assigning `isValid` here. An individual client can go from the state valid -> invalid, but should NOT go from invalid to valid.